### PR TITLE
Implement attacks.

### PR DESCRIPTION
**Pull request by staslyakhov**
Ships can now be attacked, hit, and sunk. 

See issue #6 .

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -74,12 +79,58 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
...
+			attacks.add(outcome);
+			return outcome;
+		}
+
+		int hitCount = 0;
+		for (Result r : attacks) {
+			if (r.getLocation().equals(outcome.getLocation())) {
+				outcome.setResult(INVALID);
+				return outcome;
+			} else if (r.getShip().getShipName().equals(outcome.getShip().getShipName())) {
```
Comment (KSARK) [inline src/main/java/cs361/battleships/models/Board.java]: On my end, if the `attacks` contain any `MISS`, when it runs to this line, the `null` ship will cause `NullPointerException`.

Comment (KSARK) [reply src/main/java/cs361/battleships/models/Board.java]: If you try to test this case, modify the 46 line of the `Game` class:
from
```java
opponentAttackResult.getResult() != INVALID
```
to
```java
opponentAttackResult.getResult() == INVALID
```
then the AI bug will not obstruct you.

**Comment by Csedge** [comment]
I think you can add the comment in your code, because this can help your teammate know that is the code was doing and what varible doing

File: src/main/java/cs361/battleships/models/Board.java (hunk 1 above)
Comment (staslyakhov) [reply src/main/java/cs361/battleships/models/Board.java]: Great catch! Bug should be fixed now.

**Review by KSARK — APPROVED** [review]
Looks good.

**Review by Aqcurate — APPROVED** [review]
Nice work.

**Pull request merged**